hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/HRCAssociatePrecompile.java

```diff
@@ -0,0 +1,84 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.mono.store.contracts.precompile.impl;
+
+import com.hedera.node.app.service.mono.context.SideEffectsTracker;
+import com.hedera.node.app.service.mono.contracts.sources.EvmSigsVerifier;
+import com.hedera.node.app.service.mono.fees.FeeCalculator;
+import com.hedera.node.app.service.mono.ledger.accounts.ContractAliases;
+import com.hedera.node.app.service.mono.store.contracts.WorldLedgers;
+import com.hedera.node.app.service.mono.store.contracts.precompile.InfrastructureFactory;
+import com.hedera.node.app.service.mono.store.contracts.precompile.SyntheticTxnFactory;
+import com.hedera.node.app.service.mono.store.contracts.precompile.codec.Association;
+import com.hedera.node.app.service.mono.store.contracts.precompile.utils.PrecompilePricingUtils;
+import com.hedera.node.app.service.mono.store.models.Id;
+import com.hedera.node.app.service.mono.utils.EntityIdUtils;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TransactionBody.Builder;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+import javax.inject.Provider;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+
+/**
+ * Precompile to handle the IHRC facade function for associate() function.
+ */
+public class HRCAssociatePrecompile extends AbstractAssociatePrecompile {
+
+    private final TokenID tokenID;
+    private final AccountID callerAccountID;
+
+    public HRCAssociatePrecompile(
+            final TokenID tokenID,
+            final Address callerAccount,
+            WorldLedgers ledgers,
+            ContractAliases aliases,
+            EvmSigsVerifier sigsVerifier,
+            SideEffectsTracker sideEffects,
+            SyntheticTxnFactory syntheticTxnFactory,
+            InfrastructureFactory infrastructureFactory,
+            PrecompilePricingUtils pricingUtils,
+            Provider<FeeCalculator> feeCalculator) {
+        super(
+                ledgers,
+                aliases,
+                sigsVerifier,
+                sideEffects,
+                syntheticTxnFactory,
+                infrastructureFactory,
+                pricingUtils,
+                feeCalculator);
+
+        this.tokenID = tokenID;
+        this.callerAccountID = EntityIdUtils.accountIdFromEvmAddress(callerAccount);
+    }
+
+    @Override
+    public Builder body(Bytes input, UnaryOperator<byte[]> aliasResolver) {
+        associateOp = Association.singleAssociation(callerAccountID, tokenID);
+        accountId = Id.fromGrpcAccount(Objects.requireNonNull(associateOp).accountId());
```

Comment:
Q: is it not easier to use `callerAccountID` here since it was used to create the `associateOp`?
```suggestion
        associateOp = Association.singleAssociation(callerAccountID, tokenID); // can add null check here if not in constructor
        accountId = Id.fromGrpcAccount(callerAccountID);
```